src/cls/IPM/Storage/Module.cls

```diff
@@ -85,6 +85,262 @@ Property AfterInstallMessage As %String(MAXLEN = "", XMLPROJECTION = "Element");
 
 Property Deployed As %Boolean(XMLPROJECTION = "Element");
 
+Property UpdatePackage As %Dictionary.Classname;
+
+/// Iterates through all of a module's update step classes up through the current major version and calls ApplyStepsInOrder
+Method ApplyUpdatesInOrder(
+	logFile As %Library.File = "",
+	verbose As %Boolean = 0) As %Status
+{
+    if verbose {
+        write !, "Acquiring lock on ^IPM.UpdateLock("_..Name_")"
+    }
+    $$$LockUpdateModule(..Name)
+    if ('$test) {
+        quit
+    }
+
+    set sc = $$$OK
+    set tInitTLevel = $tlevel
+    try {
+        tstart
+
+        $$$ThrowOnError(##class(%IPM.Utils.Update.VersionBase).GetUpdatePackageClassList(..UpdatePackage, .classList))
+        set keyVersion = ""
+        for {
+            // Iterate over update package class list for steps to apply. Apply all of those up to and including this major version
+            set keyVersion = $order(classList(keyVersion))
+            if (keyVersion = "") || (keyVersion > ..Version.Major) {
+                quit
+            }
+            if verbose {
+                write !, "Starting to run update steps for module="_..Name_" version="_keyVersion
+            }
+            set sc = ..ApplyStepsInOrder(keyVersion, logFile, verbose)
+        }
+
+        tcommit
+    } catch (ex) {
+        set sc = ex.AsStatus()
+    }
+    while ($tlevel > tInitTLevel) {
+        trollback 1
+    }
+
+    if verbose {
+        write !, "Releasing lock on ^IPM.UpdateLock("_..Name_")"
+    }
+    $$$UnlockUpdateModule(..Name)
+    quit sc
+}
+
+/// For a given version, apply update steps listed in the module's update package version class
+/// First run are steps that can be executed on any mirror member, followed by those that can only be run on the primary
+/// Previously executed steps will be skipped until a new one has been found.
+/// If a step has not been run, all subsequent steps will be run, regardless of if they already have or not.
+/// i.e. If an update class returns methods Foo1, Foo2, Foo3 for v1.0.0 but v1.1.0 has Foo1, Foo2, Bar1, Foo3,
+///      only Bar1, Foo3 will be run on the update for v1.1.0
+Method ApplyStepsInOrder(
+	version As %String,
+	logFile As %Library.File,
+	verbose As %Boolean = 0) As %Status [ Internal ]
+{
+    set sc = $$$OK
+    try {
+        set className = ..UpdatePackage_".V"_version
+        if '##class(%Dictionary.ClassDefinition).%ExistsId(className) {
+            if verbose {
+                write !, "Class """_className_""" does not exist. Skipping running its update steps"
+            }
+            quit
+        }
+
+        // Regardless of mirror status, run these methods first
+        set anyMirrorMemberMethodNames = $classmethod(className, "GetOrderedMethodsAnyMirrorMember")
```

Comment:
Have if $System.CLS.IsMthd(className, <method name>) for each of these to ensure we only run it if its been overridden.